generator_framework/generator.py

```diff
@@ -3,115 +3,54 @@
-                #weekday - high traffic
-                    morning = Stats(self.Morning_Increments, self._Low_Max, self._Low_Min, self._Shape, self._Function)
-                    worktime = Stats(self.Work_Hours_Increments, self._High_Max, self._High_Min, self._Shape, self._Function)
-                    evening = Stats(self.Evening_Increments, self._Low_Max, self._Low_Min, self._Shape, self._Function)
-                    # Last step is merge arrays
-                    array = np.concatenate([morning.Dist_Array, worktime.Dist_Array, evening.Dist_Array])
-                else:
-                    print("weekend - low traffic")
-                    #weekend - low traffic
-                    morning = Stats(32, self._Low_Max, self._Low_Min, self._Shape, self._Function)
-                    worktime = Stats(32, self._Low_Max, self._Low_Min, self._Shape, self._Function)
-                    evening = Stats(32, self._Low_Max, self._Low_Min, self._Shape, self._Function)
+                                      password=db_config.password, database=self._Config.Database)
+        self.Dist_Array = self._Config.Func_Type.generate()
 
-                    array = np.concatenate([morning.Dist_Array, worktime.Dist_Array, evening.Dist_Array])
-                if self.Dist_Array is None:
-                    self.Dist_Array = array
-                else:
-                    self.Dist_Array = np.concatenate([self.Dist_Array, array])
-                count += 1
-            count = 0
-        result_datetimes = np.array(pd.date_range(self._Config.Start_Date, periods=self._Increments+1, freq='15min'))[1:]
-        return np.array([result_datetimes, self.Dist_Array]).transpose()
 
-    def nparray_to_dataframe(self, nparray_data):
-        indexes = pd.DataFrame(nparray_data[:, 0])
+    def nparray_to_dataframe(self):
+        indexes = pd.DataFrame(self.Dist_Array[:, 0])
         indexes[0] = pd.to_datetime(indexes[0], format='%Y-%m-%d %H:%M:%S')
         cols = [self._Columns]
-        df = pd.DataFrame(data=nparray_data[0:, 1:],
+        df = pd.DataFrame(data=self.Dist_Array[0:, 1:],
                           index=indexes[0],
                           columns=cols)
         return df
 
-    def write_data_to_csv(self, model_name, df):
+    def write_data_to_csv(self):
+        df = self.nparray_to_dataframe()
+        model_name = self._Config.Func_Type.Name
         if not isinstance(df, pd.DataFrame):
             print("Error reading the data from database. Please test this query in Chronograf/Grafana.")
         df.to_csv(path.join(RESOURCES_DIR, model_name + "_generated.csv"))
 
-    #this will work once the io_framework can succesfully import
-    '''def write_data_to_database(self, model_name, df):
+    def write_data_to_database(self):
+        df = self.nparray_to_dataframe()
+        model_name = self._Config.Func_Type.Name
         df.to_csv(path.join(RESOURCES_DIR, model_name + "_generated.csv"))
```

Comment:
Reformat the function to follow your naming conventions. In your cases model_name is not defined so it should likely throw the error.

Reply:
Oops, ignore this comment, just noticed what you do in line 44. Renaming it still should be good, though.